lib/src/message/mod.rs

```diff
@@ -202,3 +202,41 @@ impl<'c> Message for Commit<'c> {
     }
 }
 
+
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // LineIteratorExt tests
+
+    #[test]
+    #[should_panic]
+    fn empty_message_format_check() {
+        let vec : Vec<&str> = Vec::new();
+        vec.into_iter().check_message_format().unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn empty_message_format_check2() {
+        vec![""].into_iter().check_message_format().unwrap();
+    }
+
+    #[test]
+    fn oneline_message_format_check() {
+        vec!["Foo bar"].into_iter().check_message_format().unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn malformed_message_format_check() {
+        vec!["Foo bar", "Baz"].into_iter().check_message_format().unwrap();
+    }
```

Comment:
Why are you not using `assert!()` in the tests? If things do not panic (in the above tests) they might still not return what we would expect. Especially in this test case, if `check_message_format()` panics, we get a green test for this - which is clearly not valid!

```
assert!(vec[..].into_iter().check_message_format().is_err())
```

would be the way to go here.